pandas/io/common.py

```diff
@@ -1032,34 +952,23 @@ def _maybe_memory_map(
 
     # need to open the file first
     if isinstance(handle, str):
-        if encoding and "b" not in mode:
-            # Encoding
-            handle = open(handle, mode, encoding=encoding, errors=errors, newline="")
-        else:
-            # Binary mode
-            handle = open(handle, mode)
+        # encoding will be handled later in TextIOWrapper
+        handle = open(handle, "rb")
         handles.append(handle)
 
     try:
-        # error: Argument 1 to "_MMapWrapper" has incompatible type "Union[IO[Any],
-        # RawIOBase, BufferedIOBase, TextIOBase, mmap]"; expected "IO[Any]"
-        wrapped = cast(
-            BaseBuffer,
-            _MMapWrapper(handle, encoding, errors, decode),  # type: ignore[arg-type]
-        )
-        # error: "BaseBuffer" has no attribute "close"
-        handle.close()  # type: ignore[attr-defined]
-        handles.remove(handle)
-        handles.append(wrapped)
-        handle = wrapped
-    except Exception:
-        # we catch any errors that may have occurred
-        # because that is consistent with the lower-level
-        # functionality of the C engine (pd.read_csv), so
-        # leave the file handler as is then
-        memory_map = False
+        mmap_handle = mmap.mmap(handle.fileno(), 0, access=mmap.ACCESS_READ)
+    except UnsupportedOperation:
```

Comment:
I would favor removing the try-except. A user should be informed that `mmap` failed.

Reply:
and it is not documented that mmap is meant to silently fail (at least not in read_csv).